test/test_ekfac.py

```diff
@@ -0,0 +1,1094 @@
+        model,
+        MSELoss(reduction="sum"),
+        params,
+        [(X, y)],
+    )
+
+    # save state dict
+    state_dict = ekfac.state_dict()
+
+    # create new EKFAC from state dict
+    kfac_new = EKFACLinearOperator.from_state_dict(state_dict, model, params, [(X, y)])
+
+    # check that the two EKFACs are equal
+    compare_state_dicts(ekfac.state_dict(), kfac_new.state_dict())
+    test_vec = rand(ekfac.shape[1])
+    report_nonclose(ekfac @ test_vec, kfac_new @ test_vec)
+
+
+@mark.parametrize("fisher_type", EKFACLinearOperator._SUPPORTED_FISHER_TYPE)
+@mark.parametrize(
+    "separate_weight_and_bias", [True, False], ids=["separate_bias", "joint_bias"]
+)
+@mark.parametrize(
+    "exclude", [None, "weight", "bias"], ids=["all", "no_weights", "no_biases"]
+)
+@mark.parametrize("shuffle", [False, True], ids=["", "shuffled"])
+def test_ekfac_closer_to_exact_than_kfac(
+    inv_case,
+    shuffle: bool,
+    exclude: str,
+    separate_weight_and_bias: bool,
+    fisher_type: FisherType,
+):
+    """Test that EKFAC is closer in Frobenius norm to the exact quantity than KFAC."""
+    model, loss_func, params, data, batch_size_fn = inv_case
+    params = maybe_exclude_or_shuffle_parameters(params, model, exclude, shuffle)
+
+    # Compute exact block-wise ground truth quantity.
+    linop = (
+        EFLinearOperator if fisher_type == FisherType.EMPIRICAL else GGNLinearOperator
+    )
+    exact = block_diagonal(
+        linop,
+        model,
+        loss_func,
+        params,
+        data,
+        batch_size_fn=batch_size_fn,
+        separate_weight_and_bias=separate_weight_and_bias,
+    )
+
+    # Compute KFAC and EKFAC.
+    kfac = KFACLinearOperator(
+        model,
+        loss_func,
+        params,
+        data,
+        batch_size_fn=batch_size_fn,
+        separate_weight_and_bias=separate_weight_and_bias,
+        fisher_type=fisher_type,
+        mc_samples=1_000 if fisher_type == FisherType.MC else 1,
+    )
+    kfac_mat = kfac.to_scipy() @ eye(kfac.shape[1])
+    ekfac = EKFACLinearOperator(
+        model,
+        loss_func,
+        params,
+        data,
+        batch_size_fn=batch_size_fn,
+        separate_weight_and_bias=separate_weight_and_bias,
+        fisher_type=fisher_type,
+        mc_samples=1_000 if fisher_type == FisherType.MC else 1,
+    )
+    ekfac_mat = ekfac.to_scipy() @ eye(ekfac.shape[1])
+
+    # Compute and compare (relative) distances to the exact quantity.
+    exact_norm = norm(exact)
+    exact_kfac_dist = norm(exact - kfac_mat) / exact_norm
+    exact_ekfac_dist = norm(exact - ekfac_mat) / exact_norm
+    if exclude == "weight":
+        # For no_weights the numerical error dominates.
+        assert np_allclose(exact_kfac_dist, exact_ekfac_dist, atol=1e-5)
+    else:
+        assert exact_kfac_dist > exact_ekfac_dist
+
+
+# TODO: FisherType.MC is too expensive, but could be added as optional test?
+@mark.parametrize("fisher_type", [FisherType.TYPE2, FisherType.EMPIRICAL])
```

Comment:
Same comment as above. Should we just use the MC-Fisher with the identical random seed as exact quantity? Then, the test should pass without having to crank up the number of MC samples.